framework/src/utils/MooseMeshUtils.C

```diff
@@ -291,4 +291,24 @@ isCoPlanar(const std::vector<Point> vec_pts)
   // If all the points are collinear, they are also coplanar
   return true;
 }
+
+SubdomainID
+getNextFreeSubdomainID(MeshBase & input_mesh)
```

Comment:
The assumption here is that `input_mesh` is an actual mesh object where `generate()` has been called. I don't believe this would work if the mesh hasn't been created yet (e.g. before generate() is called). Which util file should this routine be placed then?

Reply:
it could just be a class attribute in mooseMesh